src/jit/emitxarch.cpp

```diff
@@ -214,7 +217,8 @@ bool TakesRexWPrefix(instruction ins, emitAttr attr)
     // cased here.
     //
     // Rex_jmp = jmp with rex prefix always requires rex.w prefix.
-    if (ins == INS_movsx || ins == INS_rex_jmp)
+    // vpermq requires w bit to set to 1
+    if (ins == INS_movsx || ins == INS_rex_jmp || ins == INS_vpermq)
     {
```

Comment:
I don't think this is correct. vpermq is encoded with a VEX prefix (with w set to 1), not with a REX.W prefix. And this function is only asking about REX.W.